src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/ValidationHelperServiceTest.java

```diff
@@ -117,72 +107,76 @@ public void testValidateUpdateUserProfileRequestValidHappyPath() {
+        assertThat(raisedException).isInstanceOf(RequiredFieldMissingException.class);
 
         verify(auditServiceMock, times(1)).persistAudit(eq(HttpStatus.BAD_REQUEST), eq(API));
     }
 
     @Test(expected = Test.None.class)
     public void testvalidateUserStatusBeforeUpdate_scenario3() {
+        userProfile.setStatus(IdamStatus.ACTIVE);
         assertThat(sut.validateUserStatusBeforeUpdate(updateUserProfileData, userProfile, API)).isTrue();
     }
 
     @Test
     public void test_validateUserStatusBeforeUpdate_should_throw_exception_when_user_is_pending() {
-
-        doThrow(InvalidRequest.class).when(exceptionServiceMock).throwCustomRuntimeException(eq(ExceptionType.REQUIREDFIELDMISSINGEXCEPTION), any(String.class));
         final Throwable raisedException = catchThrowable(() -> sut.validateUserStatusBeforeUpdate(updateUserProfileData, userProfile, API));
-        assertThat(raisedException).isInstanceOf(InvalidRequest.class);
+
+        assertThat(raisedException.getMessage()).contains("User is PENDING or input status is PENDING and only be changed to ACTIVE or SUSPENDED for userId: null");
+        assertThat(raisedException).isInstanceOf(RequiredFieldMissingException.class);
 
         verify(auditServiceMock, times(1)).persistAudit(any(HttpStatus.class), any(ResponseSource.class));
         verify(exceptionServiceMock, times(1)).throwCustomRuntimeException(any(ExceptionType.class), any(String.class));
     }
 
     @Test
     public void test_validateUserStatusBeforeUpdate_should_throw_exception_when_user_is_pending_in_request() {
-
-        doThrow(InvalidRequest.class).when(exceptionServiceMock).throwCustomRuntimeException(eq(ExceptionType.REQUIREDFIELDMISSINGEXCEPTION), any(String.class));
         final Throwable raisedException = catchThrowable(() -> sut.validateUserStatusBeforeUpdate(updateUserProfileData, userProfile, API));
-        assertThat(raisedException).isInstanceOf(InvalidRequest.class);
+
+        assertThat(raisedException.getMessage()).contains("User is PENDING or input status is PENDING and only be changed to ACTIVE or SUSPENDED for userId: null");
+        assertThat(raisedException).isInstanceOf(RequiredFieldMissingException.class);
 
         verify(auditServiceMock, times(1)).persistAudit(any(HttpStatus.class), any(ResponseSource.class));
         verify(exceptionServiceMock, times(1)).throwCustomRuntimeException(any(ExceptionType.class), any(String.class));
     }
 
     @Test(expected = Test.None.class)
-    public void testvalidateUserPersistedWithException_scenario1() {
+    public void test_validateUserPersistedWithException_scenario1() {
         assertThat(sut.validateUserPersisted(HttpStatus.OK)).isTrue();
     }
 
-    @Test(expected = Test.None.class)
-    public void testvalidateUserPersistedWithException_scenario2() {
+    @Test(expected = ErrorPersistingException.class)
```

Comment:
wondering just changed expected exception here , how original test was passing ?

Reply:
May have been that it was not actually testing the method but returning true regardless, if you see the above test (line 174) that is the correct scenario where HttpStatus.OK should return true and not throw exception, but HttpStatus.BAD_REQUEST should throw ErrorPersistingException as it is a Bad Request